src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -948,18 +1038,24 @@ private bool IsFailOverable(AggregateException ex)
 
         private bool IsFailOverable(RequestFailedException rfe)
         {
+            var innerException = rfe.InnerException;
 
-            // The InnerException could be SocketException or WebException when endpoint is invalid and IOException if it is network issue.
-            if (rfe.InnerException != null && rfe.InnerException is HttpRequestException hre && hre.InnerException != null)
+            if (innerException == null)
             {
-                return hre.InnerException is WebException ||
-                       hre.InnerException is SocketException ||
-                       hre.InnerException is IOException;
-            }
-
-            return rfe.Status == HttpStatusCodes.TooManyRequests ||
+                return rfe.Status == HttpStatusCodes.TooManyRequests ||
                    rfe.Status == (int)HttpStatusCode.RequestTimeout ||
                    rfe.Status >= (int)HttpStatusCode.InternalServerError;
+            }
+
+            if (innerException is HttpRequestException hre && hre.InnerException != null)
```

Comment:
Seems solid, but the multiassignment to innerException threw me off a bit. I think it would help if the assignment to inner exception only happened once because it's really behaving in a branched manner.

I'd suggest

```
            if (rfe.Status == HttpStatusCodes.TooManyRequests ||
                rfe.Status == (int)HttpStatusCode.RequestTimeout ||
                rfe.Status >= (int)HttpStatusCode.InternalServerError)
            {
                return true;
            }

            Exception innerException;

            if (rfe.InnerException is HttpRequestException hre)
            {
                innerException = hre.InnerException;
            }
            else
            {
                innerException = rfe.InnerException;
            }

            // The InnerException could be SocketException or WebException when an endpoint is invalid and IOException if it's a network issue.
            return innerException is WebException ||
                   innerException is SocketException ||
                   innerException is IOException;

```